packages/dma-library/src/operations/ajna/open.ts

```diff
@@ -0,0 +1,158 @@
+import { actions } from '@dma-library/actions'
+import {
+  IOperation,
+  Protocol,
+  WithAjnaBucketPrice,
+  WithAjnaStrategyAddresses,
+  WithCollateral,
+  WithDebtAndBorrow,
+  WithFlashloan,
+  WithOptionalDeposit,
+  WithPosition,
+  WithProxy,
+  WithSwap,
+} from '@dma-library/types'
+import { NULL_ADDRESS, ZERO } from '@oasisdex/dma-common/constants'
+import { ajnaOpenOperationDefinition } from '@oasisdex/dma-deployments/operation-definitions'
+import BigNumber from 'bignumber.js'
+import { ethers } from 'ethers'
+
+type OpenArgs = WithCollateral &
+  WithDebtAndBorrow &
+  WithOptionalDeposit &
+  WithSwap &
+  WithFlashloan &
+  WithProxy &
+  WithPosition &
+  WithAjnaStrategyAddresses &
+  WithAjnaBucketPrice
+
+export async function open({
+  collateral,
+  debt,
+  deposit,
+  swap,
+  flashloan,
+  proxy,
+  position,
+  addresses,
+  price,
+}: OpenArgs): Promise<IOperation> {
+  const depositAmount = deposit?.amount || ZERO
+  const depositAddress = deposit?.address || NULL_ADDRESS
+
+  const isDepositingCollateral = deposit?.address === collateral.address
+  const isDepositingDebtTokens = deposit?.address === debt.address
+
+  const pullDebtTokensToProxy = actions.common.pullToken({
+    asset: debt.address,
+    amount: isDepositingDebtTokens ? depositAmount : ZERO,
+    from: proxy.owner,
+  })
+
+  const pullCollateralTokensToProxy = actions.common.pullToken({
+    asset: collateral.address,
+    amount: isDepositingCollateral ? depositAmount : ZERO,
+    from: proxy.owner,
+  })
+
+  const setDebtTokenApprovalOnLendingPool = actions.common.setApproval({
+    amount: flashloan.amount,
+    asset: debt.address,
+    delegate: addresses.pool,
+    sumAmounts: false,
+  })
+
+  const depositBorrow = actions.ajna.ajnaDepositBorrow({
+    pool: addresses.pool,
+    depositAmount: flashloan.amount,
+    sumAmounts: false,
+    borrowAmount: debt.borrow.amount,
+    price,
+  })
+
+  const wrapEth = actions.common.wrapEth({
+    amount: new BigNumber(ethers.constants.MaxUint256.toHexString()),
+  })
```

Comment:
move it after pull token. 
Case: collateral **IS** eth:
- wrap eth is ued, pull token is skipped
 
Case: collateral **IS NOT** eth (it's some ERC20)
- wrap eth is skipped, pull token is used